.github/workflows/build_and_deploy.yml

```
@@ -1,28 +1,36 @@
 name: Publish to GitHub Pages with Lunr Search Extension
 on:
   push:
-    branches: [main, docs]
+    branches: [ main ]
 jobs:
   build:
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: documentation/
+
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v2
-      with:
-        ref: docs
-    - name: Install Node.js
-      uses: actions/setup-node@v2
-      with:
-        node-version: '16'
-    - name: Install Antora and the Antora Lunr Extension
-      run: npm i antora @antora/lunr-extension
-    - name: Generate Site
-      run: npx antora antora-playbook.yml
-    - name: Upload GitHub Pages artifact
-      uses: actions/upload-pages-artifact@v1.0.7
-      with:
-        path:  build/site/
-        
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: docs
+      - name: Install Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+      - name: npm install
+        run: npm i
+      - name: Install Antora and the Antora Lunr Extension
+        run: npm i antora @antora/lunr-extension
+      - name: Bundle-UI
+        run: gulp
+      - name: Generate Site
+        run: npx antora antora-playbook.yml
```

I think it should be sufficient to run a single npm command here and add antora and antora/lunr-extension as dev dependency.

```suggestion
      - name: Build the site
        run: npm run "build all"
```